lib/puppet/provider/mongodb_replset/mongo.rb

```diff
@@ -238,8 +238,24 @@ def set_members
       Puppet.debug "New Hosts are: #{newhosts.inspect}"
 
       newhosts.each do |host|
+        retry_limit = 10
+        retry_sleep = 3
+
         output = {}
-        output = rs_arbiter == host ? rs_add_arbiter(host, master) : rs_add(host, master)
+        retry_limit.times do |n|
+          begin
+            output = rs_arbiter == host ? rs_add_arbiter(host, master) : rs_add(host, master)
+            if !output['ok'].zero?
+              Puppet.debug 'Host successfully added to replicaset'
+              break
+            else
+              Puppet.debug "Retry adding host to replicaset. Retry: #{n}"
+              sleep retry_sleep
+              master = master_host(alive_hosts)
+              next
```

Comment:
I think we can drop the `next` too, as it doesn't skip anything and the loop will progress to the next iteration by itself.